consensus/state.go

```diff
@@ -932,7 +932,7 @@ func (cs *ConsensusState) createProposalBlock() (block *types.Block, blockParts
 	}
 
 	// Mempool validated transactions
-	txs := cs.mempool.Reap(cs.config.MaxBlockSizeTxs)
+	txs := cs.mempool.Reap(cs.state.ConsensusParams.BlockSize.MaxTxs)
```

Comment:
Note this means this param will be set at genesis time and can't be changed. It also has an abnormally large default

Reply:
> It also has an abnormally large default

set to 10000